plugin/plugin.go

```diff
@@ -0,0 +1,43 @@
+package junoplugin
+
+import (
+	"fmt"
+	"plugin"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+//go:generate mockgen -destination=../mocks/mock_plugin.go -package=mocks github.com/NethermindEth/juno/plugin JunoPlugin
+type JunoPlugin interface {
+	Init() error
+	Shutdown() error // Todo: Currently this function will never be called.
```

Comment:
why we don't call it ?

Reply:
I've updated the PR to handle plugin.Shutdown()